agent/proxycfg/state.go

```diff
@@ -15,6 +15,11 @@ import (
 	"github.com/mitchellh/copystructure"
 )
 
+type CacheNotifier interface {
```

Comment:
Does this need to be exported, or can it be `cacheNotifier` for now?

Reply:
doesn't need to be exported